docs/source/api/v2/isos.rst

```diff
@@ -0,0 +1,124 @@
+
+``POST``
+========
+Generates an ISO from the requested ISO source.
+
+:Auth. Required: Yes
+:Roles Required: "admin" or "operations"
+:Response Type:  Object - unless the ``stream`` key is present in the request payload, in which case the actual ISO content will be returned instead of a JSON response string
+
+Request Structure
+-----------------
+:dhcp: A string that specifies whether the generated system image will use DHCP IP address leasing; one of:
+
+	yes
+		DHCP will be used, and other network configuration keys need not be present in the request (and are ignored if they are)
+	no
+		DHCP will not be used, and the desired network configuration **must** be specified manually in the request body
+
+:disk:          An optional string that names the block device (under ``/dev/``) used for the boot media, e.g. "sda"
+:domainName:    The domain part of the system image's Fully Qualified Domain Name (FQDN)
+:hostName:      The host name part of the system image's FQDN
+:interfaceMtu:  A number that specifies the Maximum Transmission Unit (MTU) for the system image's network interface card - the only valid values of which I'm aware are 1500 or 9000, and this should almost always just be 1500
+:interfaceName: An optional string naming the network interface to be used by the generated system image e.g. "bond0", "eth0", etc. If the special name "bond0" is used, an :abbr:`LACP (Link Aggregation Control Protocol)` binding configuration will be created and included in the system image
+
+	.. seealso:: `The Link Aggregation Wikipedia page <https://en.wikipedia.org/wiki/Link_aggregation>`_\ .
+
+:ip6Address:   An optional string containing the IPv6 address of the generated system image
+:ip6Gateway:   An optional string specifying the IPv6 address of the generated system image's network gateway - this will be ignored if ``ipGateway`` is specified
+:ipAddress:    An optional\ [1]_ string containing the IP address of the generated system image
+:ipGateway:    An optional\ [1]_ string specifying the IP address of the generated system image's network gateway
+:ipNetmask:    An optional\ [1]_ string specifying the subnet mask of the generated system image
+:osversionDir: The name of the directory containing the ISO source
+
+	.. seealso:: :ref:`to-api-osversions`
+
+:rootPass: The password used by the generated system image's ``root`` user
+:stream:   An optional string that must be 'yes' or 'no' (Default: no) - if it is given and is 'yes', the response payload will be the content of the ISO rather than the normal JSON response
+
+	.. note:: This is called 'stream' because it is implemented by writing the results of the ISO compression process directly into the TCP streaming socket after sending the necessary HTTP headers. As a result, this is much faster and more space-efficient than `"stream": "no"` (which will first write the entire ISO to disk, then return a URL that can be used to download it) and so it is recommended in most cases that this be "yes".
```

Comment:
Assuming non-streaming ISOs aren't present in 2.x, this note should also disappear.

Reply:
yep but not in yet will be in the isos PR